Scene heading improvements

gabornemeth opened the pull request:
Scene heading now supports lowercase/mixed and also watch for empty line after the heading.
Unit tests were already present as I see. Now "Lowercase known scene heading" test also succeeds.
This pr also includes https://github.com/bryancostanich/FountainSharp/commit/58383cb0fd01b75d0c2069719be950af4cf815b5  I have worked on heading issues on my Windows machine and master branch did not compiled (nor the current dev), so I applied this fix before started.

bryancostanich commented:
looks great. We should probably add the check for newline after heading for the other case (forced) as well.

gabornemeth commented:
I have added this check also. Also added unit tests for scene heading with no empty lines after it. There is a problem with those - the "Some action" part disappears. In the editor however it seems ok. Please check those unit tests!
I'm gonna return to this.

gabornemeth commented:
Don't bother! The unit tests I wrote were wrong. Managed to fix them. Now this pr can rest in peace.
